src/server/scripts/EasternKingdoms/TheStockade/boss_hogger.cpp

```diff
@@ -0,0 +1,208 @@
+#include "the_stockade.h"
+
+enum Spells
+{
+    SPELL_VICIOUS_SLICE = 86604,
+    SPELL_MADDENING_CALL = 86620,
+    SPELL_ENRAGE = 86736
+};
+
+enum Says
+{
+    SAY_PULL = 0, // Forest just setback!
+    SAY_ENRAGE = 1, // AreaTriggerMessage: Hogger Enrages!
+    SAY_DEATH = 2, // Yiipe!
+
+
+    WARDEN_SAY_1 = 0, // Yell - This ends here, Hogger!
+    WARDEN_SAY_2 = 1, // Say - He's...he's dead?
+    WARDEN_SAY_3 = 2, // Say - It's simply too good to be true. You couldn't have killed him so easily!
+};
+
+enum Events
+{
+    EVENT_VICIOUS_SLICE = 1,
+    EVENT_MADDENING_CALL,
+
+
+    EVENT_WARDEN_SAY_1,
+    EVENT_WARDEN_SAY_2,
+    EVENT_WARDEN_SAY_3
+};
+
+enum Points
+{
+    POINT_FINISH = 0
+};
+
+Position const wardenthelwatermovepos = { 152.019f, 106.198f, -35.1896f, 1.082104f };
+Position const wardenthelwaterpos = { 138.369f, 78.2932f, -33.85627f, 1.082104f };
+
+class boss_hogger : public CreatureScript
+{
+public:
+    boss_hogger() : CreatureScript("boss_hogger") {}
+
+    struct boss_hoggerAI : public BossAI
+    {
+        boss_hoggerAI(Creature* creature) : BossAI(creature, DATA_HOGGER) {}
+
+        void Reset() override
+        {
+            _Reset();
+            isEnraged = false;
+        }
+
+        void EnterCombat(Unit* /*who*/) override
+        {
+            _EnterCombat();
+            Talk(SAY_PULL);
+            events.ScheduleEvent(EVENT_VICIOUS_SLICE, randtime(Seconds(3), Seconds(4)));
+            events.ScheduleEvent(EVENT_MADDENING_CALL, randtime(Seconds(1), Seconds(2)));
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            Talk(SAY_DEATH);
+            _JustDied();
+            me->SummonCreature(NPC_WARDEN_THELWATER, wardenthelwaterpos);
+        }
+
+        void JustSummoned(Creature* summon) override
+        {
+            BossAI::JustSummoned(summon);
+            if (summon->GetEntry() == NPC_WARDEN_THELWATER)
+                summon->GetMotionMaster()->MovePoint(POINT_FINISH, wardenthelwatermovepos);
+
+            summons.Summon(summon);
+
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            events.Update(diff);
+
+            while (uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_VICIOUS_SLICE:
+                    DoCastVictim(SPELL_VICIOUS_SLICE);
+                    events.Repeat(randtime(Seconds(10), Seconds(14)));
+                    break;
+                case EVENT_MADDENING_CALL:
+                    DoCast(SPELL_MADDENING_CALL);
+                    events.Repeat(randtime(Seconds(15), Seconds(20)));
+                    break;
+                }
+            }
+
+            DoMeleeAttackIfReady();
+        }
+
+        void DamageTaken(Unit* /*attacker*/, uint32& /*damage*/) override
+        {
+            if (me->HealthBelowPct(30) && !isEnraged)
```

Comment:
Can you give me an example of how to use it because I did it like this: 
if (me->HealthBelowPctDamaged(30, damage))

And now he casts it everytime